packages/sdk/src/controllers/DataConnectorController.ts

```diff
@@ -0,0 +1,145 @@
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetModel(connectorId, JSON.stringify(context))
+            .then((result) => getEditorResponseData<ConnectorConfigOptions>(result));
+    };
+
+    /**
+     * All connectors have a certain set of mappings they allow to be passed into the connector methods their context. This
+     * method allows you to discover which mappings are available for a given connector. If you want to use any of these
+     * mappings, they will be available in the `context` parameter of any connector method.
+     * @param connectorId unique id of the media connector
+     * @returns connector mappings
+     */
+    getConfigurationOptions = async (connectorId: string) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetConfigurationOptions(connectorId)
+            .then((result) => getEditorResponseData<ConnectorConfigOptions>(result));
+    };
+
+    /**
+     * This method returns what capabilities the selected connector has. It gives an indication what methods can
+     * be used successfully for a certain connector.
+     * @param connectorId unique id of the Data connector
+     * @returns DataConnectorCapabilities
+     */
+    getCapabilities = async (connectorId: string) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetCapabilities(connectorId)
+            .then((result) => getEditorResponseData<DataConnectorCapabilities>(result));
+    };
+
+    /**
+     * Check if the value is a DatePropertyWrapper with the type guard
+     * @param value a dynamic value which
+     * @returns boolean value.
+     */
+    private isDatePropertyWrapper(
+        value: string | number | boolean | DatePropertyWrapper | null,
+    ): value is DatePropertyWrapper {
+        return typeof value === 'object' && value?.type === 'date';
+    }
+
+    /**
+     * Transforms an InternalDataItem into a DataItem.
+     *
+     * Converts DatePropertyWrapper values to JavaScript Date objects
+     * while keeping other values unchanged.
+     *
+     * @param dataItem - The InternalDataItem to transform.
+     * @returns The resulting DataItem with parsed date properties.
+     */
+    private mapInternalToDataItem(dataItem: InternalDataItem): DataItem {
+        const parsedItem: DataItem = {};
+
+        Object.entries(dataItem).forEach(([key, value]) => {
+            parsedItem[key] = this.isDatePropertyWrapper(value) ? new Date(value.value) : value;
+        });
+
+        return parsedItem;
+    }
+}
+
+/**
+ * Internal type to wrap the date object value on the engine side,
+ * created due to the specifics of handling dates across QuickJS ⇒ Engine ⇒ JSON ⇒ SDK.
+ */
+type DatePropertyWrapper = {
+    value: number;
+    type: 'date';
+};
+
+/**
+ * Internal (DTO) data item type,
+ * created due to the specifics of handling dates across QuickJS ⇒ Engine ⇒ JSON ⇒ SDK.
```

Comment:
Maybe something like this is clear enough?

```suggestion
 * Engine data item type.
```